lib/templateFactory.js

```diff
@@ -321,6 +341,69 @@ class TemplateFactory extends BaseFactory {
         return super.get(config);
     }
 
+    /**
+     * Get metrics for the pipelines that use the given template
+     * @method getPipelineUsage
+     * @param  {String}     fullTemplateName    Name of the template and the version or tag (e.g. chef/publish@1.2.3)
+     * @return {Promise}                        List of metrics for the pipelines or null if template is not found.
+     */
+    async getPipelineUsage(fullTemplateName) {
+        const template = await this.getTemplate(fullTemplateName);
+
+        if (!template) {
+            return null;
+        }
+        const { id } = template;
+        const metricPipelinesConfig = {
+            params: { templateId: id },
+            readOnly: true,
+            aggregationField: 'pipelineId'
+        };
+
+        // eslint-disable-next-line global-require
+        const JobFactory = require('./jobFactory');
+        const jobFactory = JobFactory.getInstance();
+
+        const pipelineIds = await jobFactory
+            .list(metricPipelinesConfig)
+            .then(pipelines => pipelines.map(p => p.pipelineId));
+
+        // eslint-disable-next-line global-require
```

Comment:
If none of the pipelines are using the template, skip rest of the logic. Add test to cover this.
```suggestion
     if(pipelineIds.length === 0) {
       return Promise.resolve([]);
     }

        // eslint-disable-next-line global-require
```